lib/tracing-limit/src/lib.rs

```diff
@@ -485,27 +703,50 @@ mod test {
             true
         }
 
-        fn on_event(&self, event: &Event<'_>, _ctx: Context<'_, S>) {
-            let mut visitor = MessageVisitor::default();
+        fn on_event(&self, event: &Event<'_>, ctx: Context<'_, S>) {
+            let mut visitor = AllFieldsVisitor::default();
             event.record(&mut visitor);
 
+            // Also capture fields from span context
+            if let Some(span) = ctx.lookup_current() {
+                for span_ref in span.scope().from_root() {
+                    let extensions = span_ref.extensions();
+                    if let Some(span_keys) = extensions.get::<RateLimitedSpanKeys>() {
+                        // Add component_id
+                        if let Some(TraceValue::String(ref s)) = span_keys.component_id {
+                            visitor.fields.insert("component_id".to_string(), s.clone());
```

Comment:
```suggestion
                            visitor.fields.insert("component_id".to_string(), s.clone());
                            break;
```

Should we break here?

Reply:
very minor nit: I also don't like how deeply this is nested but there is no other way to do this other than creating a separate function...